crates/polars-plan/src/dsl/function_expr/schema.rs

```diff
@@ -379,59 +386,55 @@ impl<'a> FieldsMapper<'a> {
     pub(super) fn map_to_date_range_dtype(
         &self,
         every: &Duration,
-        time_unit: &Option<TimeUnit>,
-        tz: &Option<String>,
+        time_unit: Option<&TimeUnit>,
+        time_zone: Option<&str>,
     ) -> PolarsResult<DataType> {
-        let inner_dtype = match (&self.map_to_supertype()?.dtype, time_unit, tz, every) {
-            #[cfg(feature = "timezones")]
-            (DataType::Datetime(tu, Some(field_tz)), time_unit, Some(tz), _) => {
-                if field_tz != tz {
-                    polars_bail!(ComputeError: format!("Given time_zone is different from that of timezone aware datetimes. \
-                    Given: '{}', got: '{}'.", tz, field_tz))
-                }
```

Comment:
@MarcoGorelli I would think that the given time zone should replace the data's time zone here. Is there a reason this throws an error?

Reply:
hey - looks like this was introduced here https://github.com/pola-rs/polars/pull/5234

sounds good to overwrite though (if clearly documented)